ivre/db/__init__.py

```diff
@@ -1251,6 +1252,25 @@ def __migrate_schema_hosts_17_18(doc):
                     )
         return doc
 
+    @staticmethod
+    def __migrate_schema_hosts_18_19(doc):
+        """Converts a record from version 18 to version 19. Version 19
+splits smb-os-discovery scripts into two, a smb-ntlm-info one that contains all
+the information the original smb-os-discovery script got from NTLM, and a
+smb-os-discovery script with only the information regarding SMB
+
+        """
+        assert doc["schema_version"] == 18
+        doc["schema_version"] = 19
+        for port in doc.get('ports', []):
+            for i, script in enumerate(port.get('scripts', [])):
+                if script['id'] == 'smb-os-discovery':
+                    smb, ntlm = xmlnmap.split_smb_os_discovery(script)
+                    port.get('scripts', [])[i] = smb
+                    port.get('scripts', []).append(ntlm)
```

Comment:
Don't do this! ;-)
You need to update the existing script, maybe using `script.update(smb)`. You don't need `port.get('scripts', [])` because if you are there, there are indeed `scripts` in `port`.
```suggestion
            for script in port.get('scripts', []):
                if script['id'] == 'smb-os-discovery':
                    smb, ntlm = xmlnmap.split_smb_os_discovery(script)
                    script.update(smb)
                    port['scripts'].append(ntlm)
```